datafusion/proto/src/bytes/mod.rs

```diff
@@ -140,6 +140,24 @@ impl Serializeable for Expr {
                     Arc::new(|| unimplemented!()),
                 )))
             }
+            fn register_udaf(
+                &mut self,
+                _udaf: Arc<AggregateUDF>,
+            ) -> Result<Option<Arc<AggregateUDF>>> {
+                unimplemented!()
```

Comment:
that goes for all `unimplemented!()` calls: this will panic, but you can totally return a proper error here `DataFusionError::NotImplemented`.

Reply:
While `unimplemented` is consistent with the code above (that also will panic) I agree it  would be better to return an error here 

perhaps a internal error would be most appropriate as this code "should never be hit"

```suggestion
                internal_err!("register_udaf called in Placeholder Registry")
```

Likewise for the locations below